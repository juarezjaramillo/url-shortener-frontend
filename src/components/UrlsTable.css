.urls-table thead tr th:first-child,
.urls-table tbody tr td:first-child {
    width: 25%;
    word-break: break-all;
}

.urls-table thead tr:first-child th {
    border-bottom: 1px solid aliceblue;
}