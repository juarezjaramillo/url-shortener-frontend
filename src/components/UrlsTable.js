import * as React from "react";
import UrlsTableRow from "./UrlsTableRow";
import "./UrlsTable.css";

class UrlsTable extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const urls = this.props.urls.map((url) =>
            <UrlsTableRow key={url.id} url={url}>
            </UrlsTableRow>
        );
        return (
            <table className="urls-table">
                <thead>
                <tr>
                    <th>Url</th>
                    <th>Title</th>
                    <th>Shortened Url</th>
                    <th>Hits</th>
                </tr>
                </thead>
                <tbody>{urls}</tbody>
            </table>
        );
    }
}

export default UrlsTable;