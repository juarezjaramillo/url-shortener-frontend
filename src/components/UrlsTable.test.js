import React from 'react';
import renderer from 'react-test-renderer';
import UrlsTable from "./UrlsTable";

it('renders correctly', () => {
    const fakeUrls = [
        {
            id:1,
            url: "https://www.google.com",
            code: "ab",
            shortened_url: "https://test-url.com/v/ab",
            hits:2,
            title:"Google Test"
        },
        {
            id:2,
            url: "https://www.google2.com",
            code: "ba",
            shortened_url: "https://test-url.com/v/ba",
            hits:1,
            title:"Google Test 2"
        },

    ]
    const tree = renderer
        .create(<UrlsTable urls={fakeUrls} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});