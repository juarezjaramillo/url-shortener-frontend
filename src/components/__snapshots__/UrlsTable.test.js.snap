// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table
  className="urls-table"
>
  <thead>
    <tr>
      <th>
        Url
      </th>
      <th>
        Title
      </th>
      <th>
        Shortened Url
      </th>
      <th>
        Hits
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        https://www.google.com
      </td>
      <td>
        Google Test
      </td>
      <td>
        <a
          className="App-link"
          href="https://test-url.com/v/ab"
        >
          https://test-url.com/v/ab
        </a>
      </td>
      <td>
        2
      </td>
    </tr>
    <tr>
      <td>
        https://www.google2.com
      </td>
      <td>
        Google Test 2
      </td>
      <td>
        <a
          className="App-link"
          href="https://test-url.com/v/ba"
        >
          https://test-url.com/v/ba
        </a>
      </td>
      <td>
        1
      </td>
    </tr>
  </tbody>
</table>
`;
